dotnet/src/Connectors/Connectors.AI.PaLM/Skills/CountTokenSkill.cs

```diff
@@ -0,0 +1,118 @@
+﻿using System;
+using Microsoft.SemanticKernel.SkillDefinition;
+using System.ComponentModel;
+using Microsoft.SemanticKernel.AI;
+using System.Net.Http;
+using Microsoft.SemanticKernel.Diagnostics;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Connectors.AI.PaLM.Skills
+{
+    public sealed class TokenSkill
+    {
+        private const string HttpUserAgent = "Microsoft-Semantic-Kernel";
+
+        private readonly string _model;
+        private readonly string? _endpoint = "https://generativelanguage.googleapis.com/v1beta2/models";
+        private readonly HttpClient _httpClient;
+        private readonly bool _disposeHttpClient = true;
+        private readonly string? _apiKey;
+
+        /// <summary>
+        /// Initializes a new instance of the Token Skill
+        /// </summary>
+        /// <param name="model">Model to use</param>
+        /// <param name="apiKey">PaLM API Key</param>
+        /// <param name="httpClient">instance of http client if already exist</param>
+        public TokenSkill(string model, string apiKey, HttpClient? httpClient = null)
+        {
+            Verify.NotNullOrWhiteSpace(model);
+            Verify.NotNullOrWhiteSpace(apiKey);
+
+            this._model = model;
+            this._apiKey = apiKey;
+            this._httpClient = httpClient ?? new HttpClient(NonDisposableHttpClientHandler.Instance, disposeHandler: false);
+            this._disposeHttpClient = false; // Disposal is unnecessary as we either use a non-disposable handler or utilize a custom HTTP client that we should not dispose.
+        }
+        /// <summary>
+        /// count token from text.
+        /// </summary>
+        /// <example>
+        /// SKContext["input"] = "hello world"
+        /// {{token.countToken $input}} => 2
+        /// </example>
+        /// <param name="input"> The string to count. </param>
+        /// <returns> The token count. </returns>
+        [SKFunction, Description("count token from text.")]
+        public async Task<int> CountToken(string input)
+        {
+            try
+            {
+                CancellationTokenSource cts = new CancellationTokenSource();
```

Comment:
No need to create CancellationTokenSource here to obtain cancellation token, just pass the CancellationToken as a parameter for the method.

Reply:
fixed